main/src/ca/uwaterloo/flix/language/phase/Lexer.scala

```diff
@@ -1089,4 +1090,35 @@ object Lexer {
     }
     TokenKind.Err(LexerError.UnterminatedBlockComment(sourceLocationAtStart()))
   }
+
+  /**
+    * Returns a new array of tokens based on the successfully lexed array of `tokens`.
```

Comment:
Doesn't actually return a new array, right? Modifies the given array and returns it.